R/config-files.R

```diff
@@ -0,0 +1,160 @@
+
+
+#' Add validation file ordering to  validation config file
+#'
+#' Impose ordering of validation child files
+#' @param filename character vector containing filenames in order
+#' @param before,after Optional destination of new filenames, default is end of existing list. Specifying both is error.
+#' @returns Used for side effect of adding validation file ordering to validation config
+#'     file. Invisibly returns TRUE on success.
+#'
+#' @rdname validation_config
+#'
+#' @examples
+#' \dontrun{
+#'
+#' vt_use_validation_config(pkg = ".")
+#'
+#' vt_add_file_to_config(filename = "myReqFile.Rmd")
+#' }
+#' @importFrom rlang abort
+#' @export
+vt_add_file_to_config <- function(filename, before = NULL, after = NULL, pkg = "." ){
+
+  validation_config <- read_validation_config(pkg = pkg)
+
+  # check whether these files are already listed
+  # handle whether filename value is list or vector
+  validation_file_list_old <- validation_config$validation_files
+  if(length(validation_file_list_old[validation_file_list_old %in% unlist(filename)]) > 0 ){
+    abort(
+      paste0(
+        "Filename(s): `",
+        paste0(filename, collapse = ", "),
+        "` ",
+        "already exists " ,
+        "validation config file. Run `valtools::vt_drop_file_from_config(filename)` first!"
+      ),
+      class = "vt.validation_config_add_already_exists")
```

Comment:
not disagreeing with this choice, but I could see an argument where it would be expected that the file would be moved, not just told the file already exists. 

I think the error message is clear, so no issues with this approach

Reply:
Yeah once we get into true relocate territory, we run into the challenge of user language for bulk activity. For example, shifting files that includes the substring "01" . I know we're trying not to have extensive dependencies - are we okay with adding {tidyselect} as a dependency?

Reply:
I'm going to merge this in now, but if we want to extend functionality of {tidyselect}, lets open an new issue